embeddings/hyperparameter_search/configspace.py

```diff
@@ -102,3 +103,28 @@ def _retrieve_embedding_type(embedding_name: str) -> str:
         else:
             raise TypeError("Embedding type not recognized")
         return embedding_type
+
+    @staticmethod
+    def _parse_yaml_params(parameters: Dict[str, Any]) -> Dict[str, Parameter]:
+        parsed_parameters = {}
+        for param_key, param_values in parameters.items():
+            param_type = param_values.pop("param_type")
```

Comment:
so about pop vs get 
Are you sure that you want to modify original object parameters provided as argument to this method? Isn't that unpleasant and unexpected side effect of calling this method? I don't like methods which do that unless there's really good reason for it

Reply:
Yes, because param_type is not expected to be passed to the Parameter's object __init__ method; it is just a way to inform which parameter will be created but should be removed for further processing. Moreover, the **parameters** object is not reused any further after the processing